### PR TITLE
Add thread safety guidelines

### PR DESCRIPTION
**Pull request by pakrym**
Fixes: https://github.com/Azure/azure-sdk-for-net/issues/5503

### CODE REVIEW

File: docs/design/dotnet/DesignGuidelines.mdk [hunk 1]
```diff
@@ -126,6 +126,10 @@ use ```Azure.Core.HttpPipeline``` to implement all service call methods.
 See [Using HttpPipeline](#dotnet-usage-httppipeline) for details.
 ~
 
+~ Must {#dotnet-method-threadsafe}
```
Comment (KrzysztofCwalina) [inline docs/design/dotnet/DesignGuidelines.mdk]: I think this should say what needs to be thread safe. In general, guideline bullets should be copy-pastable independently.

File: docs/design/dotnet/DesignGuidelines.mdk [hunk 2]
```diff
@@ -246,6 +254,16 @@ place service API in second level namespace, e.g. `Azure.KeyVault`
 consider placing model types in a ```.Models``` subnamespaces of the client types, if the model schema is very large.
 ~
 
+## Policy implementation
```
Comment (KrzysztofCwalina) [inline docs/design/dotnet/DesignGuidelines.mdk]: Would it be possible to move this section to much later in the document?

File: docs/design/dotnet/DesignGuidelines.mdk (hunk 1 above)
Comment (pakrym) [reply docs/design/dotnet/DesignGuidelines.mdk]: Done

File: docs/design/dotnet/DesignGuidelines.mdk (hunk 2 above)
Comment (pakrym) [reply docs/design/dotnet/DesignGuidelines.mdk]: Done

**Pull request merged**